packages/mesos/extra/make_disk_resources.py

```diff
@@ -116,19 +131,39 @@ def stitch(parts):
     return common
 
 
-def main(output_env_file):
+def main(output_env_file, cache_env_file):
     '''
     Find mounts and freespace matching MOUNT_PATTERN, create RESOURCES for the
     disks, and merge the list of disk resources with optionally existing
-    MESOS_RESOURCES environment varianble.
+    MESOS_RESOURCES environment variable.
 
     @type output_env_file: str, filename to write resources
     '''
     if os.path.exists(output_env_file):
         print('Volume discovery assumed to be completed because {} exists'.format(output_env_file))
         return
 
-    mounts_dfree = list(get_mounts_and_freespace(find_mounts_matching(MOUNT_PATTERN)))
+    cache_mounts = {}
+    if os.path.exists(cache_env_file):
+        print('Mesos resources cache used for mount volumes because {} exists'.format(cache_env_file))
+        with open(cache_env_file) as f:
```

Comment:
I think I've seen similar code somewhere. It could be worth to extract a function.